core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -21,6 +21,7 @@
 #include "refcount.h"
 #include "rights.h"
 #include "str.h"
+#include "wasm_export.h"
```

Comment:
Hi, the two files can be built without compilation warnings in Linux platform, you write your own Makefile to build them, and there are compilation warnings, right? It should be because that these files call `wasm_runtime_malloc`/`wasm_runtime_free`, and the prototypes are not declared.

In Linux, it uses cmake, and in core/iwasm/common/iwasm_common.cmake, it defines the macro `BH_MALLOC` and `BH_FREE` to `wasm_runtime_malloc` and `wasm_runtime_free`, see: https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/common/iwasm_common.cmake#L8-L9

And in platform_common.h, the prototypes of BH_MALLOC/BH_FREE are defined, see https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/shared/platform/include/platform_common.h#L40-L53

So could you please add the macros BH_MALLOC/BH_FREE in your own makefile like iwasm_common.cmake? In our original design, wasm_export.h is only for runtime embedder to use, we had better no use it inside runtime.